src/stream_map.rs

```diff
@@ -113,6 +114,36 @@ where
             Some((id, None)) => Poll::Ready((id, None)),
         }
     }
+
+    /// Returns an iterator over all streams whose inner type is `T`.
+    ///
+    /// If downcasting a stream to `T` fails it will be skipped in the iterator.
+    pub fn iter_typed<T>(&self) -> impl Iterator<Item = (&ID, &T)>
```

Comment:
I am not sure if `iter_typed` adds much value here. Similarly to above, the fact that we are boxing this is kind of an implementation detail, no? It is observable because they don't have to specify the type anywhere but other than that, it seems to me that `iter` should work just fine here as a name.

The compiler will complain here that you need to specify a type so it isn't like one can forget about it.

Reply:
It's not really an implementation detail, because the fact that we box the streams allows users to pass different kinds of streams in each `try_push`. And the iterator here will only return those of the pushed streams where the `F` in `try_push` is the same as the `T` here.
If we just call this `iter`, I'd expect as a user to get an iterator over all streams, or maybe a `Stream<Item= Option<T>>`, but at least that the length of the iterator equals the number of streams that were pushed.

Reply:
> If we just call this `iter`, I'd expect as a user to get an iterator over all streams, or maybe a `Stream<Item= Option<T>>`, but at least that the length of the iterator equals the number of streams that were pushed.

Even if the compiler forces you to provide a type argument?

If you call `iter::<String>()`, wouldn't it be obvious that the iterator will yield no items?

Perhaps we should call it `iter_of_type`?

Reply:
> Perhaps we should call it `iter_of_type`?

:+1: done